### PR TITLE
Update benchmark_app to pass precision via command line

### PR DESCRIPTION
**Pull request by ArtemySkrebkov-intel**
Cherry-pick of https://github.com/openvinotoolkit/openvino/pull/4167

### CODE REVIEW

**Review by ilya-lavrenov — APPROVED** [review]

**Comment by ilya-lavrenov** [comment]
@ArtemySkrebkov-intel could you please make CI green?

**Comment by akuporos** [comment]
@nkogteva what do you think, should we also add this command line argument to the python benchmark app?

**Comment by nkogteva** [comment]
@akuporos Sure. They should be aligned. I already added this remark to original PR and @ArtemySkrebkov-intel said that python will be updated during merge of this feature to master branch. So it is time)

**Comment by ArtemySkrebkov-intel** [comment]
Python benchmark_app has been updated.
@nkogteva @akuporos 
Please review.

File: inference-engine/samples/common/utils/src/args_helper.cpp [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+    return getLayout(value, supported_layouts);
+}
+
+bool isMatchLayoutToDims(InferenceEngine::Layout layout, size_t dimension) {
+    static const matchLayoutToDims_t matchLayoutToDims = {
+        {static_cast<size_t>(InferenceEngine::Layout::NCDHW), 5 },
+        {static_cast<size_t>(InferenceEngine::Layout::NDHWC), 5 },
+        {static_cast<size_t>(InferenceEngine::Layout::NCHW), 4 },
+        {static_cast<size_t>(InferenceEngine::Layout::NHWC), 4 },
+        {static_cast<size_t>(InferenceEngine::Layout::CHW), 3 },
```
Comment (nkogteva) [inline inference-engine/samples/common/utils/src/args_helper.cpp:228]: `HWC`?

File: inference-engine/samples/benchmark_app/benchmark_app.hpp [hunk 2]
```diff
@@ -108,6 +108,19 @@ static const char layout_message[] = "Optional. Prompts how network layouts shou
 // @brief message for quantization bits
 static const char gna_qb_message[] = "Optional. Weight bits for quantization:  8 or 16 (default)";
 
+// TODO: duplicate options from compile_tool
```
Comment (nkogteva) [inline inference-engine/samples/benchmark_app/benchmark_app.hpp:111]: Do we need this TODO in benchmark_app header?

File: inference-engine/tools/benchmark_tool/README.md [hunk 3]
```diff
@@ -133,7 +133,9 @@ Options:
                         graph information serialized.
   -pc [PERF_COUNTS], --perf_counts [PERF_COUNTS]
                         Optional. Report performance counters.
-
+  -ip "U8"/"FP16"/"FP32"    Optional. Specifies precision for all input layers of the network.
+  -op "U8"/"FP16"/"FP32"    Optional. Specifies precision for all output layers of the network.
+  -iop                      Optional. Specifies precision for input and output layers by name. Example: -iop "input:FP16, output:FP16". Notice that quotes are required. Overwrites precision from ip and op options for specified layers.
```
Comment (nkogteva) [inline inference-engine/tools/benchmark_tool/README.md:138]: Maybe there is a sense add test for new options to sample smoke tests

File: inference-engine/samples/common/utils/src/args_helper.cpp (hunk 1 above)
Comment (ArtemySkrebkov-intel) [reply inference-engine/samples/common/utils/src/args_helper.cpp:229]: Added.

File: inference-engine/samples/benchmark_app/benchmark_app.hpp (hunk 2 above)
Comment (ArtemySkrebkov-intel) [reply inference-engine/samples/benchmark_app/benchmark_app.hpp]: Removed.

File: inference-engine/tools/benchmark_tool/README.md (hunk 3 above)
Comment (ArtemySkrebkov-intel) [reply inference-engine/tools/benchmark_tool/README.md:138]: I will add one.

**Comment by ArtemySkrebkov-intel** [comment]
@akuporos @nkogteva 

Any comments regarding python sample itself?

**Review by nkogteva — APPROVED** [review]

File: inference-engine/samples/common/utils/src/args_helper.cpp [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+            continue;
+        }
+        if (!readArguments) {
+            continue;
+        }
+        if (args.at(i).c_str()[0] == '-') {
+            break;
+        }
+        readInputFilesArguments(files, args.at(i));
+    }
```
[the remark this answers is not in the document]
Comment (ArtemySkrebkov-intel) [reply inference-engine/samples/common/utils/src/args_helper.cpp]: Updated.

File: inference-engine/samples/common/utils/src/args_helper.cpp [hunk 5]
```diff
@@ -0,0 +1,307 @@
...
+
+void readInputFilesArguments(std::vector<std::string> &files, const std::string& arg) {
+    struct stat sb;
+    if (stat(arg.c_str(), &sb) != 0) {
+        slog::warn << "File " << arg << " cannot be opened!" << slog::endl;
+        return;
+    }
+    if (S_ISDIR(sb.st_mode)) {
+        DIR *dp;
+        dp = opendir(arg.c_str());
```
[the remark this answers is not in the document]
Comment (ArtemySkrebkov-intel) [reply inference-engine/samples/common/utils/src/args_helper.cpp]: Updated.

File: inference-engine/samples/common/utils/src/args_helper.cpp [hunk 6]
```diff
@@ -0,0 +1,307 @@
...
+            break;
+        }
+        readInputFilesArguments(files, args.at(i));
+    }
+}
+
+namespace {
+
+void splitStringList(const std::string& str, std::vector<std::string>& out, char delim) {
+    out.clear();
```
[the remark this answers is not in the document]
Comment (ArtemySkrebkov-intel) [reply inference-engine/samples/common/utils/src/args_helper.cpp]: Updated.

File: inference-engine/samples/common/utils/src/args_helper.cpp [hunk 7]
```diff
@@ -0,0 +1,307 @@
...
+            files.push_back(arg + "/" + ep->d_name);
+        }
+        closedir(dp);
+    } else {
+        files.push_back(arg);
+    }
+
+    if (files.size() < 20) {
+        slog::info << "Files were added: " << files.size() << slog::endl;
+        for (std::string filePath : files) {
```
[the remark this answers is not in the document]
Comment (ArtemySkrebkov-intel) [reply inference-engine/samples/common/utils/src/args_helper.cpp]: Updated.

**Review by pelszkow — APPROVED** [review]

**Comment by azhogov** [comment]
CI passed: https://gitlab-icv.inn.intel.com/inference-engine/product-configs/merge_requests/1404

**Pull request merged**